idseq_dag/steps/run_alignment_remotely.py

```diff
@@ -334,211 +309,78 @@ def run_chunk_wrapper(chunks_in_flight, chunk_output_files, n, mutex, target, ar
-                command.execute(
-                    command_patterns.SingleCommand(
-                        cmd="aws",
-                        args=[
-                            "s3",
-                            "cp",
-                            "--only-show-errors",
-                            multihit_local_outfile,
-                            os.path.join(self.chunks_result_dir_s3, "")
-                        ]
-                    )
-                )
-            log.write(f"finished alignment for chunk {chunk_id} on {service} server {instance_ip}")
+            log.write(f"finished alignment for chunk {chunk_id} with {self.alignment_algorithm} by lazily fetching last result")
+            return multihit_local_outfile
+
+        # TODO: (tmorse) remove compat hack https://jira.czi.team/browse/IDSEQ-2568
+        deployment_environment = os.environ.get("DEPLOYMENT_ENVIRONMENT", self.additional_attributes.get("environment"))
+        priority_name = os.environ.get("PRIORITY_NAME", "normal")
+        provisioning_model = os.environ.get("PROVISIONING_MODEL", "EC2")
+
+        index_dir_suffix = self.additional_attributes["index_dir_suffix"]
+
+        job_name = f"idseq-{deployment_environment}-{self.alignment_algorithm}"
+        job_queue = f"idseq-{deployment_environment}-{self.alignment_algorithm}-{provisioning_model}-{index_dir_suffix}-{priority_name}"
+        job_definition = f"idseq-{deployment_environment}-{self.alignment_algorithm}"
+
+        environment = [{
+            'name': f"INPUT_PATH_{i}",
+            'value': os.path.join(self.chunks_result_dir_s3, input_file),
+        } for i, input_file in enumerate(input_files)] + [{
+            'name': "OUTPUT_PATH",
+            'value': multihit_s3_outfile,
+        }]
+
+        session = boto3.session.Session()
+        client = session.client("batch")
+        response = client.submit_job(
+            jobName=job_name,
+            jobQueue=job_queue,
+            jobDefinition=job_definition,
+            containerOverrides={"environment": environment},
+            retryStrategy={"attempts": CHUNK_MAX_ATTEMPTS},
+            timeout={"attemptDurationSeconds": CHUNK_ATTEMPT_TIMEOUT}
+        )
+        job_id = response["jobId"]
+
+        log.write(f"waiting for {self.alignment_algorithm} batch queue for chunk {chunk_id}.")
+        total_timeout = CHUNK_MAX_ATTEMPTS * CHUNK_ATTEMPT_TIMEOUT
+        start = time.time()
+        end = start + total_timeout
+        delay = 5 * chunk_count  # ~1 chunk every 5 seconds to avoid throttling
+        while time.time() < end:
+            status = PipelineStepRunAlignmentRemotely._get_job_status(client, job_id)
+            if status == "SUCCEEDED":
+                break
+            if status == "FAILED":
+                log.log_event("alignment_batch_error", values={"job_id": job_id})
+                raise Exception("chunk alignment failed")
+            time.sleep(delay)
```

Comment:
If you introduce always the same delay for chunk checks, there is a significant chance that they will be all synced...  Also, one request per 5 seconds seems very conservative.